src/dkg/aggregator.rs

```diff
@@ -118,122 +118,122 @@ impl<
+    //             return Err(DKGError::EvaluationsCheckError(product.into()));
+    //         }
+    //     }
 
-        // Verify same ratio. Need this for security proof.
-        let pairs = [
-            (c_i.into(), self.config.u_1.into()),
-            (self.config.srs.g_g1.neg().into(), share.u_i_2.into()),
-        ];
-        if !E::product_of_pairings(pairs.iter()).is_one() {
-            return Err(DKGError::RatioIncorrect);
-        }
+    //     // Verify same ratio. Need this for security proof.
+    //     let pairs = [
+    //         (c_i.into(), self.config.u_1.into()),
+    //         (self.config.srs.g_g1.neg().into(), share.u_i_2.into()),
+    //     ];
+    //     if !E::product_of_pairings(pairs.iter()).is_one() {
+    //         return Err(DKGError::RatioIncorrect);
+    //     }
 
-        let powers_of_alpha = {
-            let mut current_alpha = E::Fr::one();
-            let mut powers = vec![];
-            for _ in 0..=self.participants.len() {
-                powers.push(current_alpha.into_repr());
-                current_alpha *= &alpha;
-            }
-            powers
-        };
-        let (batched_a_i, batched_g_1_neg) = {
-            let g_1_neg = self.config.srs.g_g1.neg();
-            let batched_a_i = share
-                .a_i
-                .iter()
-                .zip(powers_of_alpha.iter())
-                .map(|(a, power)| a.mul(*power))
-                .collect::<Vec<_>>();
-            let batched_g_1_neg = powers_of_alpha
-                .iter()
-                .map(|power| g_1_neg.mul(*power))
-                .collect::<Vec<_>>();
-            let mut batched_all = vec![];
-            batched_all.extend_from_slice(&batched_a_i);
-            batched_all.extend_from_slice(&batched_g_1_neg);
-            let batched_all = E::G1Projective::batch_normalization_into_affine(&batched_all);
-            let batched_a_i = batched_all[..batched_a_i.len()]
-                .into_iter()
-                .map(|x| x.clone())
-                .collect::<Vec<_>>();
-            let batched_g_1_neg = batched_all[batched_a_i.len()..]
-                .into_iter()
-                .map(|x| x.clone())
-                .collect::<Vec<_>>();
-            (batched_a_i, batched_g_1_neg)
-        };
-        // Verify evaluations are encrypted correctly.
-        let pairs = batched_a_i
-            .into_iter()
-            .zip(share.y_i.iter())
-            .zip(batched_g_1_neg.into_iter())
-            .enumerate()
-            .map::<Result<Vec<(E::G1Prepared, E::G2Prepared)>, DKGError<E>>, _>(
-                |(i, ((a, y), g_1_neg))| {
-                    let participant = self
-                        .participants
-                        .get(&i)
-                        .ok_or(DKGError::<E>::InvalidParticipantId(i))?;
-                    let pairs = vec![
-                        (g_1_neg.into(), (*y).into()),
-                        (a.into(), participant.public_key_sig.into()),
```

Comment:
`participant.public_key_sig` is in `G1` but the expected type is `E::G2Prepared`. What are the workarounds to getting this function working?